source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -362,12 +362,11 @@ Network::FilterStatus Filter::initializeUpstreamConnection() {
   }
 
   if (auto downstream_connection = downstreamConnection(); downstream_connection != nullptr) {
-    if (!read_callbacks_->connection()
-             .streamInfo()
+    if (!downstream_connection->streamInfo()
              .filterState()
-             ->hasData<Network::ProxyProtocolFilterState>(
+             .hasData<Network::ProxyProtocolFilterState>(
                  Network::ProxyProtocolFilterState::key())) {
-      read_callbacks_->connection().streamInfo().filterState()->setData(
+      getStreamInfo().filterState()->setData(
```

Comment:
`downstream_connection->streamInfo()` may be better here to keep consistent with the `if` check. 🤔

Reply:
`downstream_connection` is a const value, so we cannot invoke setData of stream info from the `downstream_connection`. The stream info from getStreamInfo is not const.

Reply:
I am not sure. 🤔 But can we just simplify the codes to something like this:

```
  auto& downstream_connection = read_callbacks_->connection();
  auto& filter_state = downstream_connection.streamInfo().filterState();
  if (!filter_state->hasData<Network::ProxyProtocolFilterState>(
          Network::ProxyProtocolFilterState::key())) {
    filter_state->setData(
        Network::ProxyProtocolFilterState::key(),
        std::make_shared<Network::ProxyProtocolFilterState>(Network::ProxyProtocolData{
            downstream_connection.connectionInfoProvider().remoteAddress(),
            downstream_connection.connectionInfoProvider().localAddress()}),
        StreamInfo::FilterState::StateType::ReadOnly,
        StreamInfo::FilterState::LifeSpan::Connection);
  }
  transport_socket_options_ =
      Network::TransportSocketOptionsUtility::fromFilterState(*filter_state);

  if (auto typed_state = filter_state->getDataReadOnly<Network::UpstreamSocketOptionsFilterState>(
          Network::UpstreamSocketOptionsFilterState::key());
      typed_state != nullptr) {
    auto downstream_options = typed_state->value();
    if (!upstream_options_) {
      upstream_options_ = std::make_shared<Network::Socket::Options>();
    }
    Network::Socket::appendOptions(upstream_options_, downstream_options);
  }
```

Reply:
Ah, got it. No need to use `downstreamConnection()`. Thanks.